### PR TITLE
Remove items Pull request

### PR DESCRIPTION
**Pull request by tamoorsaeed22**
- Removed all hardcoded items from the tasks array.

- Created a new JavaScript file for the new functionality.

- Implement a function for adding a new task

- Implement a function for deleting a task 

- Implement a function for editing task descriptions.

- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,34 +1,76 @@
...
+    UI.displayTodos(todos);
+  };
+
+  static removeTodo = (id) => {
+    let todos = Store.getTodos();
+    todos = todos.filter((todo) => todo.id !== id);
+    localStorage.setItem('todos', JSON.stringify(todos));
+    UI.displayTodos(todos);
+  };
+}
```
Comment (souad988) [inline src/index.js]: [OPTIONAL] Good job by implementing this solution , I would just suggest that you use an array that you initialize once in the constructor the same way you did in your static getStore method this way you get from localStorage only once on loading page and you just need to setItem in localStorage when adding or removing

File: src/index.js [hunk 2]
```diff
@@ -1,34 +1,76 @@
...
+  };
 
-  content = '';
-};
-displayTodos(todoList);
\ No newline at end of file
+  static generateId = () => Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+
```
Comment (souad988) [inline src/index.js]: Good job by implementing this method but in this project requirements you were asked to index list items using array length the first added item should have 1 as index the second one have 2 and so on

File: src/index.js [hunk 3]
```diff
@@ -1,34 +1,76 @@
...
+    localStorage.setItem('todos', JSON.stringify(todos));
+    UI.displayTodos(todos);
+  };
+
+  static removeTodo = (id) => {
+    let todos = Store.getTodos();
+    todos = todos.filter((todo) => todo.id !== id);
+    localStorage.setItem('todos', JSON.stringify(todos));
+    UI.displayTodos(todos);
+  };
```
Comment (souad988) [inline src/index.js]: Kindly make sure to update all indexes of the items after the one that was removed as this was clearly mentioned in the requirements => "Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).."

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @tamoorsaeed22 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f4a713832345239334a734c775a436169534c2f67697068792e676966


## Highlights
✔No linter errors
✔Good PR summary
✔Professional README

## Required Changes ♻️
_Check the comments under the review._
_ Also Please don't forget to add the update functionality as it is missing in your project and it is required for this activity 
Implement a function for editing task descriptions.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (tamoorsaeed22) [reply src/index.js]: @souad988  implementing this method but in this project requirements has been resolved by putting theminto array and fix the requirement .

File: src/index.js (hunk 3 above)
Comment (tamoorsaeed22) [reply src/index.js]: @souad988 According to the requirement updated the issue and resolved the conflicts

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @tamoorsaeed22, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

## Suggested changes :recycle: 
- [x] Nice job making the changes :clap: However, when you try to delete a task, all the tasks are deleted and the `id` of the tasks is "NaN" (see image). Please check to follow requirement.
Screenshot from 2023-05-03 12-56-42

- [x] As mentioned by the previous reviewer, kindly add a function that help to edit a task description as requested in the project requirement(see image).
Screenshot from 2023-05-03 13-08-38

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by tamoorsaeed22** [comment]
@souad988 @LionRouge1 the issue deleting and editing function issue has been resolved and the linters error are fixed as well working properly .

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**